Zend/tests/magic_methods_011.phpt

```diff
@@ -0,0 +1,11 @@
+--TEST--
+__set first argument should be a string when typed
+--FILE--
+<?php
+
+class Foo {
+	function __set(\Countable $name, $value) {}
+}
+?>
+--EXPECTF--
+Fatal error: Declaration of Foo::__set() must be compatible with Foo::__set(string $name, mixed $value): void in %s on line %d
```

Comment:
```suggestion
Fatal error: Declaration of Foo::__set() must be compatible with __set(string $name, mixed $value): void in %s on line %d
```

I would drop the class name here.

Technically, the method doesn't need to be compatible with the `: void` part, I wonder how to express that well.

Reply:
> I would drop the class name here.

The reason why I put the class name again in the second signature is to create some sort of idea of "inheritance". WDYT?

> Technically, the method doesn't need to be compatible with the `: void` part, I wonder how to express that well.

I  thought about that as well, but the only solution that I could think of was a check if the method has a return type declared and append that to the error message. WDYT?

Reply:
> The reason why I put the class name again in the second signature is to create some sort of idea of "inheritance". WDYT?

I think the format is generally good, it's just weird to see `Foo::x() must be compatible with Foo::x()`, which is a bit tautological. It would make a bit more sense to me if they aren't both exactly the same.

> I thought about that as well, but the only solution that I could think of was a check if the method has a return type declared and append that to the error message. WDYT?

Sounds like a good idea to me!